server/src/main/java/com/microsoft/java/bs/core/internal/reporter/TestReportReporter.java

```diff
@@ -92,28 +81,33 @@ private List<JvmTestOperationDescriptor> getTestStack(
     }
     // earlier check means that classname will always be found so can't have i < 0
     // reverse list order
-    List<JvmTestOperationDescriptor> result = new ArrayList<>(i + 1);
+    TestName testName = null;
     while (i >= 0) {
-      result.add(fullStack.get(i));
+      JvmTestOperationDescriptor desc = fullStack.get(i);
+      TestName currentTestName = new TestName(desc.getDisplayName(), desc.getSuiteName(),
```

Comment:
I got a feeling that we can reduce the extended property in `TestName` even more.

Basically, looks like we can merge suiteName, className and methodName to one name.

For example, we can first call `JvmTestOperationDescriptor.getJvmTestKind()`. If it's `SUITE`, then we use `JvmTestOperationDescriptor.getSuiteName()` as the name. Ortherwise, if it's `ATOMIC`, use `JvmTestOperationDescriptor.getMethodName()`, or `getClassName()` -> when `getMethodName()` is null, as the name.

It's just an idea but I'm not sure if it's still valid across different test frameworks. Just left this idea here first.

Reply:
I'll try and add `TestNG` and `ScalaTest` projects to the integ tests and see what they come up with.  Do you know of any other test engines around?

Reply:
We can first focus on junit 4, 5 and testng. Actually from the telemetry, junit 5 is kind of dominant.

There are lots of request for spock, but that's a groovy based one. Not our focus right now.

Reply:
Some test samples I used: https://github.com/microsoft/vscode-java-test/tree/main/test/test-projects/junit/src/test/java